### ISSUE TITLE
Multiple app_urls per site

### ISSUE THREAD

**Issue by drifter104** [BookStackApp/BookStack#1745]
Currently I have the app working on the following url
site1.example.com/help

I'd like to be able to have the same instance accessible from another url for example
site2.example.com/help without being redirected to site1.example.com/help

I'm using nginx as a proxy and a webserver in the following configuration.
reverseproxy (nginx) --> webserver (nginx/php) and I can "access" the site under the 2nd url but after the first request I'm redirected back to the site1.example.com/help url.

In short I'd like to set the APP_URL dynamically based on the request url. So that the url in the browser is always the one the user enters/follows a link too.

**Comment by ssddanbrown** [comment]
Hi @drifter104,
Are you able to explain the use-case in having those two different URLs pointing to a single BookStack instance?

**Comment by drifter104** [comment]
Hi (sorry for the delay)

We have different instances of our software hosted through different urls for different users. Some users are accessing through servers that give them url restrictions so they only have access to the url where their software instance is hosted (e.g. site2.example.com) and would not be able to access any content hosted on another url (e.g.site1.example.com). 

We want to be able to host the same software documentation for all users but because of the url restrictions of some users, the easiest way for us to do this would be to have the same instance working on different urls. This way all our users would have access to the same content.

**Comment by ssddanbrown** [comment]
Thanks for describing the use-case.

The support of multiple base-urls is already open under a few different issues, such as #1342, So I'll therefore close this off to reduce duplication.

**Comment by Stefansen** [comment]
I have the same issue while using bookstack app. I have no public ip address, so via internet I need to connect through FRP servers to my bookstack app. But in home I can connect bookstack app use local network through local ip.  But because of the APP_URL support only one domain, it impossible in this situation.  so if any methods can support multiple domain config.

**Comment by chriswa81** [comment]
Is it possible to specify multiple URLs for app_url? This feature would be very important to me.
